src/lua/decimal.c

```diff
@@ -81,14 +82,21 @@ ldecimal_##name(struct lua_State *L) {						\
 
 uint32_t CTID_DECIMAL;
 
-/** Push a new decimal on the stack and return a pointer to it. */
 decimal_t *
-luaT_pushdecimal(struct lua_State *L)
+luaT_newdecimal(struct lua_State *L)
 {
 	decimal_t *res = luaL_pushcdata(L, CTID_DECIMAL);
 	return res;
 }
 
+decimal_t *
+luaT_pushdecimal(struct lua_State *L, const decimal_t *dec)
+{
+	decimal_t *res = luaT_newdecimal(L);
```

Comment:
You write here

> A couple of notes around the change:
> 
> * On the first glance it seems that `luaT_pushdecimal()` is redundant,
>   because it can be written using `luaT_newdecimal()` + copying. That's
>   truth in contexts, where we know size of the `decimal_t` structure.
>   A user of the module API may decide to abstract out of particular
>   structure size and use ``luaT_pushdecimal()` with opacue `decimal_t *`
>   pointer.

But didn't we agree to push 64 bytes to lua stack, in which case it's perfectly fine to do `*lua_newdecimal(L) = *dec` or something similar?

Otherwise the user has a 64-byte decimal stab, which he might accidentally assign to a lua_newdecimal return.
IOW the user still may do `lua_newdecimal` + assignment, and this will result in writing out of bounds

Reply:
NB: The discussion was started here: https://github.com/tarantool/tarantool/pull/7498#discussion_r934386837.

Reply:
In brief, I doubt that its okay to increase pressure on the Lua memory and I doubt that this change will make `*res = *dec` assignment of `box_decimal *` values legal. If the latter is true, it becomes meaningless.

Reply:
Changed this description a bit:

> * On the first glance it seems that `luaT_pushdecimal()` is redundant,
>   because it can be written using `luaT_newdecimal()` + copying. That's
>   truth in contexts, where we know size of the internal `decimal_t`
>   structure. A user of the module API don't know it and should pass
>   `box_decimal_t *` pointer to `luaT_pushdecimal()` to write the value.

Reply:
What do we do with users who write `*lua_newdecimal(L) = *decimal` then?
This can be done now, can't it?

Reply:
Tell them about `luaT_pushdecimal()` and `box_decimal_copy()`. Maybe it is even better to make this problem more visible if we anyway don't declare `box_decimal_t` assigment as legal in the general case.

TBH, I don't know what is better and just share my doubts. If you and Vladimir are sure that it would be good to make such code working, I'll do.

Reply:
@locker Sorry, I should add you to reviewers, because here we discuss things, where you participated in the discussion in the past.

Reply:
I don't have a strong opinion regarding this one. I think we can leave everything as is (i.e. don't increase the size of memory consumed by decimals in Lua) and assume that users will use `box_decimal_copy` instead of assigning decimals directly. We can fix this in future if we get complaints. However, in general it isn't safe to copy opaque data structures so users neglecting `box_decimal_copy` have no one to blame but themselves IMO.

Reply:
Okay